### PR TITLE
Create note

### PR DESCRIPTION
**Pull request by srishtigautam21**
Add Create note functionality
Files to check:
Note.jsx
note.css
SavedNote.jsx
savedNote.css
NoteContext.js
HomePage.jsx
homePge.css

### CODE REVIEW

File: src/component/Note/Note.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+const AddNote = () => {
+  const { addNotes } = useNote();
+  const [isNoteVisible, setIsNoteVisisble] = useState(false);
+  const initialState = {
+    title: "",
+    tags: "",
+    priority: "",
+    mainContent: "",
+  };
+  const [noteContent, setNoteContent] = useState(initialState);
```
Comment (dev-prabhat) [inline src/component/Note/Note.jsx:14]: You can directly create object in this useState

File: src/context/NoteContext.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const config = {
+      headers: {
+        authorization: encodedToken,
+      },
+    };
+    try {
+      const response = await axios.post("/api/notes", { note }, config);
+      setNotes(response.data.notes);
+    } catch (e) {
+      console.log(e);
```
Comment (dev-prabhat) [inline src/context/NoteContext.js:30]: remove console before pushing code to github

**Review by dev-prabhat — APPROVED** [review]
Code is clean and readable, just mention some suggestions to work on

File: src/context/NoteContext.js (hunk 2 above)
Comment (srishtigautam21) [reply src/context/NoteContext.js:30]: This console is for errors

File: src/component/Note/Note.jsx (hunk 1 above)
Comment (srishtigautam21) [reply src/component/Note/Note.jsx:14]: Actually i have passed the initial state like this bcz after clicking on button save i want my input form to go back to initial state.

File: src/component/Note/Note.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import "./note.css";
+import { useState } from "react";
+import { useNote } from "../../context/NoteContext";
+
+const AddNote = () => {
+  const { addNotes } = useNote();
+  const [isNoteVisible, setIsNoteVisisble] = useState(false);
+  const initialState = {
```
Comment (sanjitsarkar) [inline src/component/Note/Note.jsx:8]: store the initialState into a separte file and then import and use

File: src/context/NoteContext.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import axios from "axios";
+
+const NoteContext = createContext({});
+
+const NoteProvider = ({ children }) => {
+  const [notes, setNotes] = useState([]);
+
+  useEffect(() => {
+    (async () => {
```
Comment (sanjitsarkar) [inline src/context/NoteContext.js:11]: You can separate this logic to authContext and store it in the state variable

File: src/context/NoteContext.js (hunk 2 above)
Comment (sanjitsarkar) [inline src/context/NoteContext.js:30]: remove console.log

**Review by sanjitsarkar — APPROVED** [review]

File: src/context/NoteContext.js (hunk 2 above)
Comment (srishtigautam21) [reply src/context/NoteContext.js:30]: ok will change this in next PR

File: src/component/Note/Note.jsx (hunk 3 above)
Comment (srishtigautam21) [reply src/component/Note/Note.jsx:8]: ok will do that

File: src/context/NoteContext.js (hunk 4 above)
Comment (srishtigautam21) [reply src/context/NoteContext.js:11]: i didn't got you why would we need to shift it to auth context

**Pull request merged**